docs/design_notes/setup_or_nncompact.rst

```diff
@@ -0,0 +1,78 @@
+Should I use ``setup`` or ``nn.compact``?
+=========================================
+
+In Flax's module system (named `Linen`_), submodules, parameters and/or other variables
+can be defined in two ways:
+
+1. **Explicitly** (using ``setup``):
+
+   Assign to ``self.<attr>`` inside a :meth:`setup <flax.linen.Module.setup>`
+   method. Then read ``self.<attr>`` from any "forward pass" method defined on the class.
```

Comment:
Then use the variables assigned to `self` in `setup` from any ....?

Reply:
ditto